pkg/release/release.go

```diff
@@ -306,22 +364,32 @@ next:
 	case TestAction:
 		if hr.Spec.Test.Enable {
 			logger.Log("info", "running test", "action", TestAction)
-			if err = r.test(client, hr); err != nil {
+			status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, apiV1.HelmReleasePhaseTesting)
```

Comment:
why pull all these `SetStatusPhase` calls out of the `test()` method? All else equals, seems better to leave them there, to follow the pattern used in the other methods, and in case we need to call `test()` in multiple places in the future.